boilerplate/app/components/Screen.tsx

```diff
@@ -265,19 +265,28 @@ export function Screen(props: ScreenProps) {
         style={statusBarStyle || (themeContext === "dark" ? "light" : "dark")}
         {...StatusBarProps}
       />
-
-      <KeyboardAvoidingView
-        behavior={isIos ? "padding" : "height"}
-        keyboardVerticalOffset={keyboardOffset}
-        {...KeyboardAvoidingViewProps}
-        style={[$styles.flex1, KeyboardAvoidingViewProps?.style]}
-      >
-        {isNonScrolling(props.preset) ? (
+      {/**
+       * KeyboardAvoidingView crashes in web,
+       * therefore we want to use ScrollView just for web
+       */}
+      {Platform.OS === "web" ? (
+        <ScrollView>
           <ScreenWithoutScrolling {...props} />
-        ) : (
-          <ScreenWithScrolling {...props} />
-        )}
-      </KeyboardAvoidingView>
+        </ScrollView>
```

Comment:
I'm not sure that this is a root-cause fix. I'd like to make sure we look into what's going on and why so we know that this is where we want to fix it.

Note that in the original stack trace, the root of the error actually occurs in `KeyboardAwareScrollView`.
<img width="480" alt="Screenshot 2024-10-18 at 3 23 49 PM" src="https://github.com/user-attachments/assets/58ce098a-8b20-47cd-b5b7-b50945622d1a">

Replacing KeyboardAwareScrollView also stops the issue from occurring:
```
function ScreenWithScrolling(props: ScreenProps) {
  ...
  useScrollToTop(ref)
  const MyComponent = Platform.OS === "web" ? ScrollView : KeyboardAwareScrollView

  return (
    <MyComponent
      bottomOffset={keyboardBottomOffset}
      {...{ keyboardShouldPersistTaps, scrollEnabled, ref }}
```

Also note that on the original issue @frankcalise linked this ticket where someone noted that adding `import "setimmediate"` also appears to fix the issue. I tried this out and it appears to work, as well (`setimmediate` is apparently brought in by `fbjs` which is a dependency of `react-native-web`.

I'm not opposed to an immediate fix to unblock things, but I think we definitely need to dig deeper into this to find and fix root cause.

I looked at the error in the dev console and it gives a clearer stack trace of the issue: that `internal.js` file is part of `react-native-keyboard-controller`. its use of setImmediate makes this not compatible with web.
<img width="595" alt="Screenshot 2024-10-18 at 3 39 45 PM" src="https://github.com/user-attachments/assets/a1fc1c6b-9c9c-4b34-a164-a685277f98ad">

I think the more appropriate fix is to swap out `KeyboardAwareScrollView` for a regular scrollview, since it wasn't designed for web.

Reply:
In addition, I think we should probably use a const to define a new component wrapping `KeyboardAwareScrollView ? ScrollView` like we did with the `useEffect` below, and export it from a file like it's a component. that way other components that may want to use this can get a version compatible with web, instead of everyone having to write a platform ternary.